src/services/goToDefinition.ts

```diff
@@ -171,13 +171,15 @@ namespace ts.GoToDefinition {
         if (!baseDeclaration) return;
 
         const baseTypeNode = getEffectiveBaseTypeNode(baseDeclaration);
-        const baseType = baseTypeNode ? typeChecker.getTypeAtLocation(baseTypeNode) : undefined;
-        if (!baseType) return;
+        if (!baseTypeNode) return;
+        const expression = skipParentheses(baseTypeNode.expression);
+        const base = isClassExpression(expression) ? expression.symbol : typeChecker.getSymbolAtLocation(expression);

```

Comment:
Actually, why are you checking for `isClassExpression`? Can't you use `getDeclaredTypeOfSymbol`?

Reply:
I have to follow `Node -> Symbol -> Type`, and `getSymbolAtLocation` doesn't support (parenthesised) class expressions right now, although `getTypeAtLocation` does. I happen to know that class expressions have a symbol, though, so I can special-case the `Node -> Symbol` step for now.

I'm going to follow this PR with one that moves the paren/class-expr check into getTypeAtLocation, but it changes lots of our baselines since, apparently, we use it for type baselines.